### PR TITLE
adding size options

### PR DESCRIPTION
**Pull request by hairmot**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -19,14 +19,26 @@ function parse(req) {
 
     let baton = {
       label: query.label || ((query.compression ? 'gzip ' : '') + 'size'),
-      color: query.color || 'brightgreen',
+      color: query.color || null,
```
Comment (ngryman) [inline index.js]: If the user does not want the `test` feature, the badge must have a default color.

File: index.js [hunk 2]
```diff
@@ -19,14 +19,26 @@ function parse(req) {
...
       style: query.style || null,
+      pass: query.pass || query.fail || Infinity,
+      fail: query.fail || query.pass || Infinity,
```
Comment (ngryman) [inline index.js]: I would slightly modify the API.

I would prefer having a `max` (or `limit`) parameter which defines a hard limit. If the size is below `max`, the badge becomes `red`, if not you don't change anything (as the default color is already green).

The idea of having a *blurry* zone where the badge becomes yellow is a good idea, but it should be an opt-in.
Either we could have a `softlimit` or a `threshold` for that. If present and if the size is above `threshold` and below `limit`, then the badge becomes yellow.

What do you think?

File: index.js [hunk 3]
```diff
@@ -19,14 +19,26 @@ function parse(req) {
...
       value: 'unknown',
       extension: 'svg',
       size: 0,
       compression: query.compression,
       compressedSize: 0,
-      err: null
+      err: null,
+      updateColor: function() {
+        if (!this.color) {
```
Comment (ngryman) [inline index.js]: We would check if `limit` is present here instead.

**Comment by hairmot** [comment]
Sounds good. Have implemented changes as `max `and `threshold`

File: index.js [hunk 4]
```diff
@@ -21,12 +21,24 @@ function parse(req) {
       label: query.label || ((query.compression ? 'gzip ' : '') + 'size'),
       color: query.color || 'brightgreen',
       style: query.style || null,
+      max: query.max || query.threshold || null,
+      threshold: query.threshold || query.max || null,
```
Comment (ngryman) [inline index.js]: With this API `max` and `threshold` are aliases. It's not clear.
`max` as a hard limit, and an optional `threshold` can be added to set a soft limit.

Basically:
```js
{
  max: query.max || null,
  threshold: query. threshold || null
}
```

File: index.js [hunk 5]
```diff
@@ -21,12 +21,24 @@ function parse(req) {
...
+        if (this.max) {
+          let thresh = [this.max, this.threshold, this.size].sort((a, b) => a - b)
+          switch (this.size) {
+            case (thresh[0]) : this.color = 'brightgreen' ; break
+            case (thresh[1]) : this.color = 'yellow' ; break
+            case (thresh[2]) : this.color = 'red' ; break
+          }
+        }
+      }
     }
```
Comment (ngryman) [inline index.js]: `baton` is just a POJO, and does not contain any function. Can you put this function in the global scope as other operations (ie. `fetch`, `compressed`, ...)?

Also I would simplify the implementation a bit:
```js
function updateColor(baton) {
  if (!baton.max) return baton

  if (baton.size > baton.max) {
    baton.color = 'red'
  }
  else if (baton.threshold && baton.size > baton.threshold) {
    baton.color = 'yellow'
  }

  return baton
}
```

File: index.js [hunk 6]
```diff
@@ -74,6 +86,7 @@ function fetch(baton) {
     }).then(res => {
       baton.size = Number(res.headers['content-length'])
       baton.data = res.body
+      baton.updateColor()
```
Comment (ngryman) [inline index.js]: This would be removed in favor of a global function called in the promise chain later.

**Comment by ngryman** [comment]
Hey thanks, just a few more tweaks and we are good to go 👍

**Comment by hairmot** [comment]
cool! please see changes as requested

**Comment by ngryman** [comment]
Awesome! Thank you very much 🙏

**Comment by ngryman** [comment]
Ugh for some reason, `contributor-faces` didn't add you. I'll fix that.

**Pull request merged**